include/triton/core/tritonbackend.h

```diff
@@ -580,6 +603,39 @@ TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_ResponseOutput(
     const char* name, const TRITONSERVER_DataType datatype,
     const int64_t* shape, const uint32_t dims_count);
 
+/// Create an output state tensor. The returned state object is only valid
+/// before the TRITONBACKEND_RequestStateUpdate is called. The state should not
+/// be freed by the caller.
+///
+/// \param response The request.
+/// \param state Returns the new response output state.
+/// \param name The name of the output tensor.
+/// \param datatype The datatype of the output tensor.
+/// \param shape The shape of the output tensor.
+/// \param dims_count The number of dimensions in the output tensor
+/// shape.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_RequestStateNew(
+    TRITONBACKEND_Request* request, TRITONBACKEND_State** state,
+    const char* name, const TRITONSERVER_DataType datatype,
+    const int64_t* shape, const uint32_t dims_count);
+
+/// Update the implicit state stored for each sequence. If the number of state
```

Comment:
So, the state here is a collection of states described in the protobuf? Looks bit strange..

Reply:
No, the state here is very similar to `TRITONBACKEND_ResponseOutput`. The protobuf description is only used for determing the shape and I/O mapping.